### ISSUE TITLE
Query about validation statistics

### ISSUE THREAD

**Issue by Imogen-D** [kr-colab/locator#29]
Hi,

Thank you so much for your software.

I just have a quick question about the values output for mean and median validation error on the log file.
Where do these values come from? Is there any ability to get the whole range of estimated values for the validation dataset, and also to see which samples were used for validation?

I had thought the values were from the history.txt file, but when I manually calculate the mean and median from this file it varies from that output on the log file. Therefore, related to this question, what are the columns in the history.txt file representing?  

Looking forward to hearing from you

**Comment by cjbattey** [comment]
Hi Imogen, 

The validation loss metrics are calculated here in the `predict_locs` function (https://github.com/kr-colab/locator/blob/master/scripts/locator.py#L283). The variable naming is confusing (sorry) -- `testgen` is actually the genotype matrix of the "validation" samples, and `predgen` is the genotype matrix of the "test" samples with truly unknown locations. These two lines get the predicted locations of the validation set and then un-normalize them back to their original dimensions (there's a z-normalization step earlier for the coordinates earlier in the pipeline): 
```
p2=model.predict(testgen) #print validation loss to screen
p2=np.array([[x[0]*sdlong+meanlong,x[1]*sdlat+meanlat] for x in p2])
```
(https://github.com/kr-colab/locator/blob/master/scripts/locator.py#L300)

To get more info on the validation set you could add a line after #301 to write the object `p2` out to file with something like `pd.DataFrame(p2).to_csv("file_name")` and that should let you check the whole range. 

The `history.txt` file records the validation loss (mean euclidean distance) calculated on the normalized coordinates, so the scale is different from the `mean/median` messages printed to the screen after a run, which are calculated after un-normalizing the data.
